### DISCUSSION TITLE
Are there known problems with python environments / instable behaviour?

### DISCUSSION THREAD

**Discussion by dlr-jk** [Spine-project/Spine-Toolbox#1732] [Discussion]
Hi there,

in the course of our TradeRES activities for coupling the models AMIRIS and EMLab, we came across some weird behaviour:
- When running a tool in isolation and specifying a working python interpreter, this worked fine in the first place.
- After a couple runs, the execution of this tool all of a sudden stopped working and instead, an error was thrown within one of the packages we use (that is obviously functional).
- Spinetoolbox has been installed in another environment and started from there.

Spine Toolbox Environment:
- python 3.18.3
- spinetoolbox 0.6.6.dev0

I'm sorry not to be able to deliver a better error report since I haven't really noted down all the steps I made.
The fix has been simply reverting everything by discarding changes to our work made on the python-scripts and workflows. So I cannot tell for sure that this was SPINE-related or some usage error. Sorry for that!

I was wandering,
- whether anyone has come across a similar thing.
- if there are known issues with problems in managing python environments or the particular version of SPINE.

Thank you and best regards,
Johannes

**Comment by soininen** [comment]
I'm not aware of any known issues similar to what is described here. That's not to say there aren't any ;) In any case, if you encounter such problems in the future, please try to mark down exact steps on how to reproduce them and possible Tracebacks/error messages you might get. We developers need a way to reproduce the issue before we can fix it.

0.6.6.dev0 sounds a bit old (though we haven't been updating the version number too often). There is an update available on `pip` but due to quick development of Toolbox I recommend [installation from sources using Git](https://github.com/Spine-project/Spine-Toolbox#installation-from-sources-using-git).

**Comment by dlr-jk** [comment]
Thank you for the very quick response. I've updated the spinetoolbox version successfully.
In the meantime, I have experienced some other weird behaviour, not only within my spine environment, but also for other python environments, which indicates that there might be something wrong with my overall mambaforge python installation. So I guess, the behaviour I mentioned above was not SPINE-related.

Also, when developping the workflow further, we found out that it makes more sense to define an overall python interpreter including all the dependencies (there were no conflicts between our required dependencies) and we had a look into how you store the information in JSON files which you can edit rightaway as an alternative for using the GUI.

After digging a bit more into it, I have to give complements for this very professional and appealing tool! :-)

**Reply by soininen** [reply]
It is good to hear you managed to get further in developing your workflow! Was there something missing in the GUI that made you edit the JSON files by hand? It is just that we would like to keep the freedom to change the structure of those files without breaking anyone's workflows.
